front/src/applications/stdcm/components/StdcmResults/SendToRailwayManagerModal.tsx

```diff
@@ -0,0 +1,430 @@
+  const [comment, setComment] = useState('');
+  const [csCode, setCsCode] = useState('');
+  const diffMinutes = msToMin(scheduledDepartureTime.getTime() - departureTime.getTime());
+
+  const baseToleranceBefore = firstStep.tolerances.before.total('minute');
+  const baseToleranceAfter = firstStep.tolerances.after.total('minute');
+
+  const [beforeTolerance, setBeforeTolerance] = useState(
+    (baseToleranceBefore + diffMinutes).toString()
+  );
+  const [afterTolerance, setAfterTolerance] = useState(
+    (baseToleranceAfter - diffMinutes).toString()
+  );
+  const [pathTypeError, setPathTypeError] = useState(false);
+  const pdfBlobRef = useRef<Blob | null>(null);
+
+  const [sendLastMinuteRequest, { isLoading, isSuccess }] =
+    osrdRailwayManagerApi.useSendLastMinuteRequestSendLastMinuteRequestPostMutation();
+
+  const convoyDetails = useMemo(
+    () => [
+      { label: t('modal.compositionCode'), value: consist?.speedLimitByTag ?? '-' },
+      {
+        label: t('modal.totalTonnage'),
+        value: consist?.totalMass ? `${consist.totalMass} t` : '-',
+      },
+      { label: t('modal.rollingStock'), value: consist?.towedRollingStock?.name },
+      { label: t('modal.referenceEngine'), value: stdcmData.rollingStock.name ?? '-' },
+      {
+        label: t('modal.maxSpeed'),
+        value: consist?.maxSpeed != null ? `${consist.maxSpeed} km/h` : '-',
+      },
+      {
+        label: t('modal.totalLength'),
+        value: consist?.totalLength != null ? `${consist.totalLength} m` : '-',
+      },
+      { label: t('modal.gauge'), value: consist?.loadingGauge ?? '-' },
+    ],
+    [consist, stdcmData.rollingStock.name, t]
+  );
+
+  const pathTypeOptions = [
+    { value: 'LongFRET', label: t('modal.LongFRET') },
+    { value: 'Autre', label: t('modal.Autre') },
+  ];
+
+  const pdfDocument = useMemo(
+    () => (
+      <StdcmSimulationReportSheet
+        stdcmLinkedTrains={linkedTrains}
+        stdcmData={stdcmData}
+        consist={consist}
+        simulationReportSheetNumber={simulationReportSheetNumber}
+        operationalPointsList={operationalPointsList}
+        simulationSheetLogo={simulationSheetLogo}
+        similarTrains={similarTrains}
+      />
+    ),
+    [
+      linkedTrains,
+      stdcmData,
+      consist,
+      simulationReportSheetNumber,
+      operationalPointsList,
+      simulationSheetLogo,
+      similarTrains,
+    ]
+  );
+
+  const handleCsChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const input = e.target.value.toUpperCase();
+    setCsCode(input);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const transformedValue = e.target.value
+      .replace(/&/g, 'et') // replace "&" with "et"
+      .replace(/</g, '('); // replace "<" with "("
+
+    setComment(transformedValue);
+  };
+
+  const handleBeforeChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
```

Comment:
If the tolerances are now numbers, you just need to do the conversion here, and in `validateTolerance`, you need to check that the value is not NanN
```suggestion
    const value = Number(e.currentTarget.value);
```

Reply:
Done